skbio/alignment/_tabular_msa.py

```diff
@@ -715,8 +792,92 @@ def reindex(self, key=None, keys=None):
             keys_.flags.writeable = False
 
         self._keys = keys_
+        self._minter = minter
 
     @experimental(as_of='0.4.0-dev')
+    def append(self, sequence, minter=None):
```

Comment:
And I'm assuming if keys do exist, and no minter or key is provided to append, then that would also be an error? ie we can never have an MSA where some sequences have keys and others don't?